lib/mirage_block_partition.ml

```diff
@@ -45,19 +47,20 @@ module Make(B : Mirage_block.S) = struct
     let sector_start = Int64.add sector_start b.sector_start in
     let sector_end = Int64.add sector_start num_sectors in
     sector_start >= b.sector_start && sector_end <= b.sector_end
+      let connected = true (* or false, depending on your needs *)
 
   let read b sector_start buffers =
     (* XXX: here and in [write] we rely on the underlying block device to check
        for alignment issues of [buffers]. *)
-    if is_within b sector_start buffers
-    then
+ if connected && is_within b sector_start buffers then
```

Comment:
Use `b.connected` - that is, the record member `connected` of `b`. We should return a different error than `` `Out_of_bound`` if the device is disconnected.

Reply:
@reynir thank you for the suggestions. really helped. i have pushed the latest changes, kindly review them